shop4All-backend/src/main/java/com/example/shop4All_backend/util/ScheduledTask.java

```diff
@@ -0,0 +1,36 @@
+package com.example.shop4All_backend.util;
+
+import com.example.shop4All_backend.entities.Product;
+import com.example.shop4All_backend.repositories.ProductRepo;
+import com.example.shop4All_backend.services.ProductService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@Component
+public class ScheduledTask {
+
+    @Autowired
+    ProductRepo productRepo;
+
+    @Autowired
+    ProductService productService;
+
+    //Scheduled task to run daily at 12 AM to check and remove expired discounts
+    @Scheduled(cron = "0 0 0 * * *")
+    @Async
+    public void task1() {
```

Comment:
Try to rename this method based on what you are trying to achieve (or what you have implemented). As a developer, I should be able to understand what this method does not only through your implementation, but also by the name of the method.